### ISSUE TITLE
[BUG] Error importing WSDL from untrusted HTTPS endpoint

### ISSUE THREAD

**Issue by sebastian-cofi** [Axway-API-Management-Plus/apimanager-swagger-promote#226]
**Swagger-Promote version**
1.6.4
> Please make sure to use the latest Swagger-Promote version.

**API-Manager and Service-Pack Version**
7.6.2 SP3

**Expected behavior**
API should be imported without any error

**Actual behavior**
Exception while importing the API

**Log-Output in Debug**
```
------------------------------------------------------------------------
API-Manager Promote: 1.6.3-2 - I M P O R T

To report issues or get help, please visit:
https://github.com/Axway-API-Management-Plus/apimanager-swagger-promote
------------------------------------------------------------------------

0      INFO  nvironmentProperties| Trying to load environment properties from file: env.properties ... not found.
872    INFO  IImportConfigAdapter| Loading API-Definition from: https://url.domain.tld/webservices/SOAP_HTTP_Service?wsdl
970    ERROR                  App| Cannot validate/fulfill configuration file.
com.axway.apim.lib.AppException: Cannot validate/fulfill configuration file.
        at com.axway.apim.swagger.APIImportConfigAdapter.getDesiredAPI(APIImportConfigAdapter.java:237)
        at com.axway.apim.App.run(App.java:196)
        at com.axway.apim.App.main(App.java:45)
Caused by: com.axway.apim.lib.AppException: Cannot load API-Definition from URI: https://url.domain.tld/webservices/SOAP_HTTP_Service?wsdl
        at com.axway.apim.swagger.APIImportConfigAdapter.getAPIDefinitionFromURL(APIImportConfigAdapter.java:646)
        at com.axway.apim.swagger.APIImportConfigAdapter.getAPIDefinitionAsStream(APIImportConfigAdapter.java:591)
        at com.axway.apim.swagger.APIImportConfigAdapter.getAPIDefinitionContent(APIImportConfigAdapter.java:573)
        at com.axway.apim.swagger.APIImportConfigAdapter.getDesiredAPI(APIImportConfigAdapter.java:221)
        ... 2 more
Caused by: javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
        at sun.security.ssl.Alerts.getSSLException(Alerts.java:192)
        at sun.security.ssl.SSLSocketImpl.fatal(SSLSocketImpl.java:1946)
        at sun.security.ssl.Handshaker.fatalSE(Handshaker.java:316)
        at sun.security.ssl.Handshaker.fatalSE(Handshaker.java:310)
        at sun.security.ssl.ClientHandshaker.serverCertificate(ClientHandshaker.java:1639)
        at sun.security.ssl.ClientHandshaker.processMessage(ClientHandshaker.java:223)
        at sun.security.ssl.Handshaker.processLoop(Handshaker.java:1037)
        at sun.security.ssl.Handshaker.process_record(Handshaker.java:965)
        at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:1064)
        at sun.security.ssl.SSLSocketImpl.performInitialHandshake(SSLSocketImpl.java:1367)
        at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1395)
        at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1379)
        at org.apache.http.conn.ssl.SSLConnectionSocketFactory.createLayeredSocket(SSLConnectionSocketFactory.java:396)
        at org.apache.http.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:355)
        at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:142)
        at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:373)
        at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:394)
        at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:237)
        at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:185)
        at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
        at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
        at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:72)
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:221)
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:165)
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:140)
        at com.axway.apim.swagger.APIImportConfigAdapter.getAPIDefinitionFromURL(APIImportConfigAdapter.java:643)
        ... 5 more
Caused by: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
        at sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:397)
        at sun.security.validator.PKIXValidator.engineValidate(PKIXValidator.java:302)
        at sun.security.validator.Validator.validate(Validator.java:262)
        at sun.security.ssl.X509TrustManagerImpl.validate(X509TrustManagerImpl.java:324)
        at sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:229)
        at sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:124)
        at sun.security.ssl.ClientHandshaker.serverCertificate(ClientHandshaker.java:1621)
        ... 27 more
Caused by: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
        at sun.security.provider.certpath.SunCertPathBuilder.build(SunCertPathBuilder.java:141)
        at sun.security.provider.certpath.SunCertPathBuilder.engineBuild(SunCertPathBuilder.java:126)
        at java.security.cert.CertPathBuilder.build(CertPathBuilder.java:280)
        at sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:392)
        ... 33 more

```
**Test case sample**
> Please describe the steps you have done to replicate the issue

execute this command below :

```
su - api -c 'export JAVA_HOME=/home/api/apigateway/Linux.x86_64/jre/ && /home/api/install/software/swagger-promote/scripts/api-import.sh -c /home/api/install/conf/swagger/service/service-config.dev.json -h localhost -u apiadmin -p xxxxxxxxxxx -f true'
```
**Notes**
I obfuscated the hostname and the service name.

**Comment by cwiechmann** [comment]
Hi @sebastian-cofi, I didn't had the time to replicate the issue, however based on a code-review I made a few changes:
- accept all server certificates 
- don't perform any host-validation against the provided certificate

Can you please try to replicate your WSDL with the attached Swagger-Promote version?

[apimanager-swagger-promote-1.6.5-SNAPSHOT.zip](https://github.com/Axway-API-Management-Plus/apimanager-swagger-promote/files/4102415/apimanager-swagger-promote-1.6.5-SNAPSHOT.zip)

**Comment by sebastian-cofi** [comment]
Hi @cwiechmann the attached version 1.6.5-SNAPSHOT is working fine.

**Comment by cwiechmann** [comment]
Great. The change will be released with version 1.6.5
